### PR TITLE
Added ts types

### PR DESCRIPTION
**Pull request by maghis**
Hey, I've added Typescript types, looking for feedback.

### CODE REVIEW

**Comment by maghis** [comment]
@mcollina feedback?

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": "Generate and Parse CoAP packets",
   "main": "index.js",
   "scripts": {
+    "tstest": "tsc -p ./ts",
     "test": "./node_modules/.bin/mocha --bail --reporter spec --check-leaks"
```
Comment (mcollina) [inline package.json]: can you run those tests after the mocha ones?

File: package.json [hunk 2]
```diff
@@ -26,8 +27,12 @@
...
+    "@types/node": "7.0.11",
     "chai": "~1.8.0",
-    "mocha": "~1.13.0"
-  }
+    "mocha": "~1.13.0",
+    "pre-commit": "0.0.3",
+    "typescript": "2.2.2"
+  },
+  "types": "index.d.ts",
+  "typings": "index.d.ts"
```
Comment (mcollina) [inline package.json]: do we need those two? why both?

File: package.json (hunk 1 above)
Comment (maghis) [inline package.json]: Sure, happy to :)

File: package.json (hunk 2 above)
Comment (maghis) [inline package.json]: Support for types has been introduced in TS 2.0, a lot of people still use https://github.com/typings/typings, this makes it compatible with both.

**Pull request merged**